### PR TITLE
#4 #8 #6 Setup Gradle, Kotlin, Spring Boot Reactive, Swagger, Ktlint, Detekt, GitHub Workflow

### LINKED ISSUE #4: Create a minimal Spring Boot Kotlin backend service
## Description

To start with we need to create a project that just prints hello world and some dummy passing test. Also github workflow must be setup.

### LINKED ISSUE #6: Add minimal OpenAPI spec to backend service.
## Description

After minimal backend will be ready let's immediately and OpenAPI specification and setup codegen to stick with it.

### LINKED ISSUE #8: Add evil formatters and linters
## Description

Find a Kotlin formatter and some amount of linters to ensure that obvious mistakes are filtered by a machine. Also as we are not so familiar with Kotlin, it will help us to write a cleaner code.

### PR DESCRIPTION
**Pull request by vityaman**

### CODE REVIEW

File: .github/workflows/markdown.yml [hunk 1]
```diff
@@ -1,11 +1,6 @@
 ---
 name: Markdown Check
 on:
-  push:
-    paths:
-      - ".github/workflows/*"
-      - ".markdownlint.yaml"
-      - "**/*.md"
```
Comment (vityaman) [inline .github/workflows/markdown.yml:8]: A Pull Request is the only way to push to the main branch, so "on pull request" is sufficient.

File: gateway/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        extendsFrom(configurations.annotationProcessor.get())
+    }
+}
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    implementation("org.springframework.boot:spring-boot-starter-webflux")
```
Comment (vityaman) [inline gateway/build.gradle.kts:34]: Required to use kotlin coroutines in rest controllers

File: gateway/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    implementation("org.springdoc:springdoc-openapi-starter-webflux-ui:2.4.0")
```
Comment (vityaman) [inline gateway/build.gradle.kts:35]: Required to show swagger page.

File: gateway/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+}
+
+tasks.withType<Test> {
+    useJUnitPlatform()
+}
+
+val generatedDir = "${layout.projectDirectory}/build/generated"
+val generateControllers = "generateControllers"
+
+tasks.register<GenerateTask>(generateControllers) {
```
Comment (vityaman) [inline gateway/build.gradle.kts:61]: The task to generate rest controllers from OpenAPI Spec.

**Comment by vityaman** [comment]
Going to add an evil detekt config tomorrow.
- [x] [Default config](https://github.com/detekt/detekt/blob/main/detekt-core/src/main/resources/default-detekt-config.yml)
- [x] [Integrate with GitHub code scanning](https://detekt.dev/docs/introduction/reporting#integration-with-github-code-scanning)  (skip)
- [x] [Baseline.xml](https://detekt.dev/docs/introduction/baseline) (skip)
- [x] [Go through rulesets](https://detekt.dev/docs/rules/comments)

**Comment by vityaman** [comment]
@lulalend, the MR is ready to be reviewed!

**Comment by Andryss** [comment]
Spring Boot?

Spring boot -> Ktor ?

**Review by lulalend — APPROVED** [review]
Vitya, everything is wonderful at first glance! Thanks

**Pull request merged**